src/output.c

```diff
@@ -332,10 +350,7 @@ output_config_apply(struct server *server,
 
 		/* Only do Layout specific actions if the commit went trough */
 		if (need_to_add) {
-			wlr_output_layout_add_auto(server->output_layout, o);
-			output->scene_output =
-				wlr_scene_get_scene_output(server->scene, o);
-			assert(output->scene_output);
+			add_output_to_layout(server, output);
```

Comment:
Don't we need to destroy the scene output in the `need_to_remove` branch? E.g. `wlr_scene_output_destroy()`?

Or as alternative, we could try `wlr_scene_get_scene_output()` in `add_output_to_layout()` and only fall back to create a new one if returns `NULL`.

Reply:
Agree on all points, it's a weird API and a comment would be helpful.

Reply:
I added a comment and also requested wlroots clarify their documentation: https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4357

Reply:
If https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4358 gets merged I think we should make `struct output` the owner of the scene_output. E.g. creating it in `new_output_notify()` and destroy it in `output_destroy_notify()`.

That should make stuff much cleaner.

Reply:
For when that is merged: https://github.com/jlindgren90/labwc/commit/f8405250f45cd559d0e65d9200f6d0332c47d01a

Reply:
I realized it also works to just call `wlr_scene_output_destroy()` explicitly before `wlr_output_layout_remove()` (somewhat like you originally suggested). If we destroy it first, then `wlr_output_layout_remove()` doesn't. So it ends up correct whether or not https://gitlab.freedesktop.org/wlroots/wlroots/-/merge_requests/4358 gets merged.